doc/conf.py

```diff
@@ -96,17 +96,15 @@ class ExecDirective(Directive):
 
     def run(self):
         try:
-            code = inspect.cleandoc(
-                """
-            def usermethod():
-                {}
-            """
-            ).format("\n    ".join(self.content))
-            exec(code)
-            result = locals()["usermethod"]()
+            # clean the content, then put back into a list
+            cleancode = inspect.cleandoc("\n".join(self.content)).split("\n")
+            code = "def usermethod():\n    " + "\n    ".join(cleancode)
+            globals = {}
+            exec(code, globals)
 
-            if result is None:
+            result = globals["usermethod"]()
```

Comment:
I'm really not sure why the usermethod no longer shows up in `locals`.

I just tried globals on a whim and it was there. Is this dangerous?

Reply:
I don't know why this is happening. I played with it.

Do we need to clean the code for this to work?

Reply:
I don't know why this is happening. I played with it.

Is there a reason we are rewriting the `inspect.cleandoc` stuff?